src/calliope/backend/where_parser.py

```diff
@@ -17,6 +17,7 @@
 from calliope.exceptions import BackendError
 
 if TYPE_CHECKING:
+    from calliope import config
     from calliope.backend.backend_model import BackendModel
```

Comment:
I'd prefer if it was moved to a regular import. A conflict is a code smell that indicates that the software could be arranged better, and this may hide those cases.
They are a useful annoyance.